src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc

```diff
@@ -66,6 +66,8 @@ struct alts_tsi_handshaker {
   bool shutdown = false;
   // Maximum frame size used by frame protector.
   size_t max_frame_size;
+  // The list of preferred transport protocols.
+  std::optional<absl::string_view> preferred_transport_protocols;
```

Comment:
This is dangerous because the buffer that underlies the `absl::string_view` is not guaranteed to persist for the lifetime of the `alts_tsi_handshaker` instance. Can we can this to `std::optional<std::string>` instead?